test/python/circuit/test_circuit_operations.py

```diff
@@ -895,6 +896,26 @@ def test_reverse_bits_with_registers(self):
 
         self.assertEqual(qc.reverse_bits(), expected)
 
+    def test_reverse_bits_with_registerless_bits(self):
+        """Test reversing order of registerless bits."""
+        q0 = Qubit()
+        q1 = Qubit()
+        c0 = Clbit()
+        c1 = Clbit()
+        qc = QuantumCircuit([q0, q1], [c0, c1])
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.x(0).c_if(1, True)
+        qc.measure(0, 0)
+
+        expected = QuantumCircuit([c1, c0], [q1, q0])
+        expected.h(1)
+        expected.cx(1, 0)
+        expected.x(1).c_if(0, True)
+        expected.measure(1, 1)
+
+        self.assertEqual(qc.reverse_bits(), expected)
```

Comment:
This is a good test.  Please could you also add some tests where there's a mix of registers and registerless bits, and some where there are some registers that overlap with each other?

Reply:
I added three additional test cases. They succeeded with the new logic. The downside may be that the way of constructing the `expect` circuit is not very straightforward, especially in the cases of overlapping bits.